packages/presentation/src/editor/ContentEditor.tsx

```diff
@@ -52,8 +52,8 @@ export function ContentEditor(props: {
         <StateLink
           {...props}
           state={{
-            id: mainDocumentState!.document!._id,
-            type: mainDocumentState!.document!._type,
+            id: mainDocumentState?.document?._id,
+            type: mainDocumentState?.document?._type,
```

Comment:
The `!` here actually caused a bug, [as React Compiler hoists these properties](https://playground.react.dev/#N4Igzg9grgTgxgUxALhASwLYAcIwC4AEwBA6gIYwB2alA5gPJR4A2NCAknBJQQL4EAzGBAwEA5AAEwZangCeAejRdKYMQB1q2XIWIAhCAA8ANAQDCFACamAYswQmCAFQeF+QkeKky08hVDQNLRx8IgIoMAQLZmYAIzI4AGtTCIQAWQQMCFN5LAQCAAknNIAZAAVhLDAcuTyCACUEBLwAUXsMBEo3QWFRMRgmuDwgzBDdAgBlPDI8BBKaRL4ezzFpWUVhJgQYEe1Q4iwByK6ZtG4SiDgyewA5Mg6wLAT8917xADp3pQBGAA5KXZjIiaAgEWgIPAVBAANzQCAA7gA1a5QBAkXwACxs1ziCWSIIIUNhCIsMGsBImPnkABEEAIyFBmJCBsT4cYCU4YDIwMwZgh2TxwpEJnAMZkyALQalOdzeXgzpRJQRcvkyjMMUqVZMqXJqZcoB0ugLXitPkouttKNc1JpRjplbV8sACWkyDQ9XADZ08FM+UqocdpvLuBMmvAMWquRhqhS8DAoENYAgPV6umrKAhI-cY5QTX0zSqbcF7cQU4a8PMwJCZC9lvmFGXvZXqxnASWCI20zXmEsPPXOy2EMw26FUkS4fDfbM6x9-JFxwipwggpoHECBFBKEMFeZuLMui1LL5cAAKQ4QKrIYGCyz68vsSwAfivVZgNFoBNvnvLTkdz4Ir7vgSGBupQA5Lv+rrune3pLgS3A2PqYBqngGJXme6pXihGIAJQEAAvAAfAQ0IQGgljwZQUzxomAxZtG6FPFGYBXtRCZ4EmA7ppmFDZnhREkWRFGCgMAgsWEAD65EvnG74ANwEBJKoyW+dB8AA2gAugSr7sUm9HiWxtHJjBXYZgZOlhqKBlYUc3qnCGVkRrx0aaLwOFXo0zRtJk3rXqCKhVv5oIEF+qZ4A+SqgmFP6OlFBAgdB36wUG-IEqCCFIdh8UhnGel0S5OYhQQolFSFkQUNZhXxbpxkGUq-D4QQ55VJoBKBeM3RNdKXKqHKCpnnZJzBpQFxXLc9wII8zw4e13BBWAorigRQoICKYogSes2UHNqiEFBYGmRWCwrak0S4kkJ7pQQg0XuJRSlBUd0ADwPSUACCW4YrgPnloR-HEc6grFQMHFUDd13Fc9S7zJQiSQ8VRCfO8LVgLwCPFVWfL4cAQOI-jBDSQloHgalACE7wxd6FNSeSwME6CynE0l4VLhTVNdDTKrxQzEkVeGNkEPQsQAFYIEM7zem+U0nvzVXMThPOI7w6P08VCiEQj23Fbw8XqYlh3JV0S6mHLznMZpSrbbtC1LSBv51N1kQZFkV301tBHEYtG1kO84J4CeBuk3yj6U0dodKY6BAAD7R+IYDfcuOFk3rQdHSbAF22QlsEtbgodc1LITkup3zkXi6pYHJPp6locc3gEfkTHcdiGIptZw7CB5wF82EIcMITqX6SZBAJ4ewJeMhWgAg3WTadGz6tdhwveGg7APCUIyzBtfTa-g27+PQzqtL0oyzIDwiGNI58-sLkiKJopi2IxPEl2Twz0IP1e8+s+Ty-hWTAgStipYCgLEVgCcECWCvP3VkS4UZgIgWKOmDNopcgEHgGB5dJypUpugvAwC+A4VVgzb24pO44zIfbR0JCCZYw4mAHGV9QTPVJJYZqZBLBHjoDjb4-AuRHgiDjAATPwBOnCIDwl4bwTWasCbPRcIYQgGAtjsLAGgAAXggHGAAGfg8IEBoFoBiPA+F1AgA6IIjA5jZGoMRsAAOYgDYAFp67vF5LEIcYhiHMJCs9BQii8C2NQf4thwT8a0PVuEgg2s+CmH1tXBeGcqFkE7jkUwsDi6pU0t3QmM8Tz1wfAQAAZMU0KR1O54XfiVCE68IZyOelxbsV9CmWBxq0yJiN64UOAN0mhV9MqemQuqHGgyIjYU6cVXKNEwY8WYqMqieU6qFUmeVJyBlKHrJWVfWqsyNnAF2fpbZciNbXViarAke8eAHz8XYBwoU0ADG3NwMxIAuDMANDtEAgh7CGGkQQMURiTGvIEGgGINiEbAB-uWEupT6kM1YVYDhXD3w4wAMz8DwNwbRULEm-z5P-csBBHziBoGKN8wwCBXjEFcJgCoxAyN8bilmMLcH12JfCuxBBnp3zYb44qEkJKbixuAhAEkBBIXqEBORDMFAKAIFNVgXRXFoGkKK5xGYlHOKVfkCQhY4BviwHgZxiqaB4AUJQCAJrDBYFYHAXwziZByH5SFMgjDgAHQHLDRIqyGaWBmGQR1jDzFkHMS60EAi0BCOAKI8NAFNGovMbEXAlhthhplQTLFGZXlku2L4dNXKCDRLscATJCJfWI38byqwxbEZ4XQi6xFZJkXcNoCI-hnCo3utEcqbFObKDkvzSAWtCK7mGDBGQLA6LGUZoRQYcdAhfmvMtRmCFs6QmBPjVo6RI7C3cvIFQd8jAWBsE4NwAgpz10IoCa4Xd17513vkfOn5Dgd1xqhpu9R27gB8MfXYhRvUeR8nfZmnGeAK17sJn8SgABpBAchXkuLcRgVV6i6DvFmEogtkGCZcG0BmLo7riBmAgKm6lXBU1iD4BBvdn8PlTRxgcTCzNDb4tmCjdU1GQNRO49ym9Si-2VoUA+uN-ix2CdCTW3xPiM2SbJLWmTCNGlHWbNxK+0KUrY2Zax1lfIK3TPynM7MCyjJ7OOaQrZ8yDmWezBW0qONSqrMvVDBQ4nc5uU0CAXgQA):
```ts
if (
    $[0] !== mainDocumentState.document._id ||
    $[1] !== mainDocumentState.document._type ||
    $[2] !== searchParams
  ) {
    t0 = (props_0) => (
      <StateLink
        {...props_0}
        state={{
          id: mainDocumentState.document._id,
          type: mainDocumentState.document._type,
          _searchParams: Object.entries(searchParams),
        }}
      />
    );
    $[0] = mainDocumentState.document._id;
    $[1] = mainDocumentState.document._type;
    $[2] = searchParams;
    $[3] = t0;
  } else {
    t0 = $[3];
  }
```
Of course, this doesn't work when `document` or `mainDocumentState` are null: `mainDocumentState.document._id`.
[By changing these lines to use optional chaining the problem is solved](https://playground.react.dev/#N4Igzg9grgTgxgUxALhASwLYAcIwC4AEwBA6gIYwB2alA5gPJR4A2NCAknBJQQL4EAzGBAwEA5AAEwZangCeAejRdKYMQB1q2XIWIAhCAA8ANAQDCFACamAYswQmCAFQeF+QkeKky08hVDQNLRx8IgIoMAQLZmYAIzI4AGtTCIQAWQQMCFN5LAQCAAknNIAZAAVhLDAcuTyCACUEBLwAUXsMBEo3QWFRMRgmuDwgzBDdAgBlPDI8BBKaRL4ezzFpWUVhJgQYEe1Q4iwByK6ZtG4SiDgyewA5Mg6wLAT8917xADp3pQBGAA5KXZjIiaAgEWgIPAVBAANzQCAA7gA1a5QBAkXwACxs1ziCWSIIIUNhCIsMGsBImPnkABEEAIyFBmJCBsT4cYCU4YDIwMwZgh2TxwpEJnAMZkyALQalOdzeXgzpRJQRcvkyjMMUqVZMqXJqZcoB0ugLXitPkouttKNc1JpRjplbV8sACWkyDQ9XADZ08FM+UqocdpvLuBMmvAMWquRhqhS8DAoENYAgPV6umrKAhI-cY5QTX0zSqbcF7cQU4a8PMwJCZC9lvmFGXvZXqxnASWCI20zXmEsPPXOy2EMw26FUkS4fDfbM6x9-JFxwipwggpoHECBFBKEMFeZuLMui1LL5cAAKQ4QKrIYGCyz68vsSwAfivVZgNFoBNvnvLTkdz4Ir7vgSGBupQA5Lv+rrune3pLgS3A2PqYBqngGJXme6pXihGIAJQEAAvAAfAQ0IQGgljwZQUzxomAxZtG6FPFGYBXtRCZ4EmA7ppmFDZnhREkWRFGCgMAgsWEAD65EvnG74ANwEBJKoyW+dB8AA2gAugSr7sUm9HiWxtHJjBXYZgZOlhqKBlYUc3qnCGVkRrx0aaLwOFXo0zRtJk3rXqCKhVv5oIEF+qZ4A+SqgmFP6OlFBAgdB36wUG-IEqCCFIdh8UhnGel0S5OYhQQolFSFkQUNZhXxbpxkGUq-D4QQ55VJoBKBeM3RNdKXKqHKCpnnZJzBpQFxXLc9wII8zw4e13BBWAorigRQoICKYogSes2UHNqiEFBYGmRWCwrak0S4kkJ7pQQg0XuJRSlBUd0ADwPSUACCW4YrgPnloR-HEc6grFQMHFUDd13Fc9S7zJQiSQ8VRCfO8LVgLwCPFVWfL4cAQOI-jBDSQloHgalj7vDF3rk1J5LAwToLKcTSXhRBFNHdTKrxfTEkVeGNkEPQsQAFYIEM7zem+U0nrzVXMThXOI7w6N08VCiEQj23Fbw8XqYlh3JV0S6mDLznMZpSrbbtC1LSBv51N1kQZFkV101tBHEYtG1kO84J4Ceeuk3y5OU10HOOgQAA+EfiGA33LjhACEOsB0dRsATbZDmwSluCh1zUshOS6nfOBeLql-sk6nZNswbeDU+RkfR2IYjGxndsIDnAXzYQhwwhOxfpJkEAnm7Al4yFaACDdCcp7XrMh3geGg7APCUIyzBtXTy-gy7+PQzqtL0oyzJ9wiGNI58vsLkiKJopi2IxPEl3j-T0K31es8s6lCc1+FCcEArYqWAoCxFYLHBAlgry91ZEuFGICwFilpvTaKXIBB4CgaXScqUKaoLwIAvgOFlb009uKduOMSG20dEQgmWMOJgBxufUEz1SSWGamQSwR46A42+PwLkR4Ig4wAEz8FjuwiA8JuG8HVirAmz0XCGEIBgLYrCwBoAAF4IBxgABn4PCBAaBaAYjwPhdQIAOj8IwKY6RyDEbAD9mIPWABaBe7xeSxCHGIQhjCQrPQUPIvA1jkG+JYYE-G1DVahIIJrPgphdaVznqlVuXt245FMNAwuqVNKd0JlPE8C8HwEAAGSFNCkddueEX4lQhCvCGMjnpcW7OffJlgcbNPCYjBeZDgCdKoefTKnpkLqhxv0iI2F2nFVyjRMGPFmLDKonlOqhVxnlScgZchqylnn1qtMtZwBtn6U2TItW11onKwJNvHgu8fF2AcKFNAAxtzcBMSALgzADQ7RAIIewhhJEEDFAYoxzyBBoBiFYhGwBP7liLsU2p9NmFWDYRw98OMADM-A8DcE0RC+JX8+S-3LAQR84gaBijfMMAgV4xBXCYAqMQUjvHYuZlC7BC9CWwpsQQZ618WHeOKhJCSm4sagIQBJAQSF6hARkfTBQCgCBTVYF0ZxaBpDCscRmBRjiFX5AkIWOAb4sB4EcfKmgeAFCUAgEawwWBWBwF8I4mQcheUhTIPQ4AB0Byw0SMs+mlgZhkHtfQ0xZBTFOtBHwtAAjgDCNDQBdRyLTGxFwJYbYIapUEwxRmZ5JLti+FTRyggkSbHAHSQib1iNfHcqsIWxGeF0JOvhWSRFnDaBCN4ewiNrrhHKkxVmygpLc0gGrXCm5hgwRkCwKi+laa4UGFHQIb5zzzUZjBdOoJ-jY0aMkUO-NnLyBUHfIwFgbBODcAIMc1dcK-GuG3Ze2dN7ZGzq+Q4LdMaobrtUZu4APD702Lkb1HkfJX3ppxngMtO7CZ-EoAAaQQHIZ5TiXEYGVaoug7xZgKLzeBgmXBtAZi6K64gZgIDJspVwZNYg+BgZ3W-N5U0cYHEwkzfWuLZgo3VJRoDETOOcqvQon95aFB3pjb4kd-HglVu8V4tN4myTVqkwjepR1mzcXPpClK2NGXMeZXyMtkz8ozOzHMoyOzDnEI2bMvZ5nsxltKjjUqyzz1QwUKJ7OblNAgF4EAA) 👍